src/iop/toneequal.c

```diff
@@ -1677,24 +1678,12 @@ void gui_update(struct dt_iop_module_t *self)
 void gui_changed(dt_iop_module_t *self, GtkWidget *w, void *previous)
 {
   dt_iop_toneequalizer_gui_data_t *g = (dt_iop_toneequalizer_gui_data_t *)self->gui_data;
-  if(w == g->method     ||
-     w == g->blending   ||
-     w == g->feathering ||
-     w == g->iterations ||
-     w == g->quantization)
-  {
-    invalidate_luminance_cache(self);
-  }
-  else if (w == g->details)
-  {
-    invalidate_luminance_cache(self);
+  if (w == g->details)
     show_guiding_controls(self);
-  }
   else if (w == g->contrast_boost || w == g->exposure_boost)
-  {
-    invalidate_luminance_cache(self);
     dt_bauhaus_widget_set_quad_active(w, FALSE);
-  }
+
+  invalidate_luminance_cache(self);
```

Comment:
So you invalidate the cache on any GUI interaction ? That's not needed for the settings only changing exposure (EQ graphs and sliders in simple view).

Reply:
My bad, I misinterpreted the old code there. Restored